assets/src/dashboard/components/pageStructure/index.js

```diff
@@ -172,34 +188,68 @@ export function LeftRail() {
         </Content>
         <Content>
           <NavList>
-            {enabledPaths.map((path) => (
-              <NavListItem key={path.value}>
-                <NavLink
-                  active={path.value === state.currentPath}
-                  href={resolveRoute(path.value)}
-                  aria-label={
-                    path.value === state.currentPath
-                      ? sprintf(
-                          /* translators: %s: the current page, for example "My Stories". */
-                          __('%s (active view)', 'web-stories'),
-                          path.label
-                        )
-                      : path.label
-                  }
-                  {...(path.isExternal && {
-                    rel: 'noreferrer',
-                    target: '_blank',
-                    onClick: () => onExternalLinkClick(path),
-                  })}
-                >
-                  {path.label}
-                </NavLink>
-              </NavListItem>
-            ))}
+            {enabledPrimaryPaths.map((path) => {
+              const isActive = path.value === state.currentPath;
+              const Icon = isActive ? path.IconActive : path.Icon;
+
+              return (
+                <NavListItem key={path.value}>
+                  <NavLink
+                    active={path.value === state.currentPath}
+                    href={resolveRoute(path.value)}
+                    aria-label={
+                      path.value === state.currentPath
+                        ? sprintf(
+                            /* translators: %s: the current page, for example "My Stories". */
+                            __('%s (active view)', 'web-stories'),
+                            path.label
+                          )
+                        : path.label
+                    }
+                    {...(path.isExternal && {
+                      rel: 'noreferrer',
+                      target: '_blank',
+                      onClick: () => onExternalLinkClick(path),
+                    })}
+                  >
+                    <NavLinkContent active={path.value === state.currentPath}>
+                      {Icon ? <Icon width="22px" /> : <div />}
```

Comment:
is the div here doing anything if there's no icon?

Reply:
It's using grid css so the empty div fills the space. There's probably a better way to do this though

Reply:
oh okay. makes sense. that's fine by me. was just curious

Reply:
Could also handle this by having the link content start at the second column of the grid by using [grid-template-areas](https://css-tricks.com/snippets/css/complete-guide-grid/). That'd require a bit of a restructure.